src/openrct2/ride/Ride.cpp

```diff
@@ -1926,6 +1914,15 @@ static void ride_music_update(Ride* ride)
     OpenRCT2::RideAudio::UpdateMusicInstance(*ride, rideCoords, sampleRate);
 }
 
+static void ride_music_update(Ride* ride)
+{
+    const auto& rtd = ride->GetRideTypeDescriptor();
+
+    if (!rtd.HasFlag(RIDE_TYPE_FLAG_MUSIC_ON_DEFAULT) && !rtd.HasFlag(RIDE_TYPE_FLAG_ALLOW_MUSIC))
+        return;
+    rtd.MusicUpdateFunction(ride);
```

Comment:
When the RTD is converted to an object this property will become an enum (probably) so perhaps it would make sense to do:
```cpp
switch(rtd.MusicUpdateFunction)
{
    case MusicUpdateFunction::Circus:
      CircusMusicUpdate(ride);
      break;
    case MusicUpdateFunction::Default:
       DefaultMusicUpdate(ride);
       break;
}```

Reply:
The switch case adds a very small performance overhead compared to the function pointer I believe... Also the TrackPaintFunction will check against a string when importing/exporting to json, I would do the same for the MusicUpdateFunction. It's not the same with the OperatingMode as the value is checked at multiple places in the code.

Reply:
I would seriously doubt it would make a difference but I'm not overly fussed for now just think it might simplify things further down the line.